rtos/TARGET_CORTEX/mbed_rtos_rtx.c

```diff
@@ -52,7 +52,7 @@ MBED_NORETURN void mbed_rtos_start()
     _main_thread_attr.cb_size = sizeof(_main_obj);
     _main_thread_attr.cb_mem = &_main_obj;
     _main_thread_attr.priority = osPriorityNormal;
-    _main_thread_attr.name = "main_thread";
+    _main_thread_attr.name = "rtx_main";
```

Comment:
Main thread should probably keep its name as this is actually where the `main()` starts.